### PR TITLE
Brew tap floc 1611

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes FLOC-1611

### CODE REVIEW

File: docs/indepth/installation.rst [hunk 1]
```diff
@@ -84,17 +84,10 @@ Make sure Homebrew has no issues:
...
-
-   alice@mercury:~$ brew tap ClusterHQ/tap
-   ...
-   alice@mercury:~$ brew install flocker-|latest-installable|
-   ...
-   alice@mercury:~$ brew test flocker-|latest-installable|
-   ...
-   alice@mercury:~$
+.. task:: test_homebrew flocker-|latest-installable|
+   :prompt: alice@mercury:~$
```
Comment (jongiddy) [inline docs/indepth/installation.rst]: Not clear to me why the `flocker-|latest-installable|` argument is provided?  It does not appear to be used in the `task_test_homebrew`?

File: flocker/provision/_sphinx.py [hunk 2]
```diff
@@ -66,7 +71,11 @@ def run(self):
         task = getattr(tasks, 'task_%s' % (self.arguments[0],))
         prompt = self.options.get('prompt', '$')
         if len(self.arguments) > 1:
-            task_arguments = self.arguments[1].split()
+            # Some tasks can include the latest installable version as (part
+            # of) an argument. This replaces a placeholder with that version.
+            latest = get_installable_version(version)
+            task_arguments = [item.replace(PLACEHOLDER, latest) for
+                              item in self.arguments[1].split()]
```
Comment (jongiddy) [inline flocker/provision/_sphinx.py]: This repeats the code in `flocker.docs.version_extensions.VersionCodeBlock`. Maybe make this a separate function.  But again, not seeing why this is necessary?

File: docs/indepth/installation.rst (hunk 1 above)
Comment (adamtheturtle) [inline docs/indepth/installation.rst]: `flocker-|latest-installable|` is the argument for the `recipe_url` parameter. Perhaps that parameter name should be improved and a docstring added there. This let's the task render as:

```
brew install flocker-0.4.0
```

for example.

File: flocker/provision/_sphinx.py (hunk 2 above)
Comment (adamtheturtle) [inline flocker/provision/_sphinx.py]: Explained why it is necessary in a comment above. I will try making it a separate function and then judge if it is nicer that way.

Comment (jongiddy) [inline flocker/provision/_sphinx.py]: Can you please point to an example where this is done?  Why does the homebrew task get passed the argument, but does not use it?

Comment (adamtheturtle) [inline flocker/provision/_sphinx.py]: The homebrew task does use the argument:

```python
def task_test_homebrew(recipe_url):
    return [
        Run(command="brew update"),
        Run(command="brew install {url}".format(url=recipe_url)),
        Run(command="brew test {url}".format(url=recipe_url)),
    ]
```

It uses the argument to install and test the recipe - the argument is named `recipe_url` but can really just be the name of the recipe if that repository is tapped. The argument is used in `commands = task_test_homebrew(recipe_url)` in `admin.testbrew.main`. The argument is also used in the new code:

`.. task:: test_homebrew flocker-|latest-installable|`

That task will render the commands from  `task_test_homebrew` as if the `recipe_url` were e.g. `flocker-0.3.3dev8` if 0.3.3dev8 is the latest installable package.

File: docs/gettinginvolved/infrastructure/release-process.rst [hunk 3]
```diff
@@ -365,7 +365,7 @@ Release
         export VERSION=0.1.2
```
Comment (jongiddy) [inline docs/gettinginvolved/infrastructure/release-process.rst]: This creates two sections of code (the export and the task), with no intervening text. It looks strange.

Comment (jongiddy) [inline docs/gettinginvolved/infrastructure/release-process.rst]: Is it possible to use the latest-installable argument here as well, to allow removal of the export?

Comment (adamtheturtle) [inline docs/gettinginvolved/infrastructure/release-process.rst]: I will put some intervening text in.

It is not possible - `|latest-installable|` on `master` (which doc-dev is built from, doc-dev is used for the release) uses the latest tag. When the docs are built the tag will be that of the previous release.

**Comment by jongiddy** [comment]
Tidy up the section in the release process documentation to remove the consecutive code extracts, and this is OK to merge (now that I understand what is happening with `latest-installable`

**Pull request merged**